src/main/java/org/assertj/core/api/recursive/comparison/DualValue.java

```diff
@@ -170,16 +170,35 @@ public boolean isExpectedFieldAnOrderedCollection() {
   }
 
   public boolean isActualFieldAnIterable() {
-    // ignore Path to be consistent with isExpectedFieldAnIterable
-    return actual instanceof Iterable && !(actual instanceof Path);
+    // ignore Path and ValueNode to be consistent with isExpectedFieldAnIterable
+    return actual instanceof Iterable && !(actual instanceof Path || isActualFieldAJsonValueNode());
   }
 
   public boolean isExpectedFieldAnIterable() {
     // Don't consider Path as an Iterable as recursively comparing them leads to a stack overflow, here's why:
     // Iterable are compared element by element recursively
     // Ex: /tmp/foo.txt path has /tmp as its first element
     // so /tmp is going to be compared recursively but /tmp first element is itself leading to an infinite recursion
-    return expected instanceof Iterable && !(expected instanceof Path);
+    // Don't consider ValueNode as an Iterable as they only contain one value and iterating them does not make sense.
+    return expected instanceof Iterable && !(expected instanceof Path || isExpectedFieldAJsonValueNode());
+  }
+
+  private boolean isActualFieldAJsonValueNode() {
+    return isAJsonValueNode(actual);
+  }
+
+  private boolean isExpectedFieldAJsonValueNode() {
+    return isAJsonValueNode(expected);
+  }
```

Comment:
I'd suggest dropping these intermediate methods.

Reply:
I replicated what was done for `isAnOrderedCollection` but totally fine for me to remove them.